src/main/java/org/saltyrtc/client/keystore/Box.java

```diff
@@ -86,4 +86,11 @@ public boolean equals(Object o) {
         return Arrays.equals(this.data, other.getData())
             && Arrays.equals(this.nonce, other.getNonce());
     }
+
+    @Override
+    public int hashCode() {
+        int result = Arrays.hashCode(nonce);
+        result = 31 * result + Arrays.hashCode(data);
+        return result;
+    }
```

Comment:
I think `Box` is just an internal API and it doesn't need to be hashable. What's the advantage of overriding `hashCode` in this case? Also, what's the `31` magic number?

Reply:
You're right. It does not hurt though.

It's good practice to implement the `hashCode()` method when there's also an implemented `equals()` method.

This code is generated by the IDE. The `31` is just a popular number to use: https://stackoverflow.com/a/299748